source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/imu/imu.py

```diff
@@ -0,0 +1,197 @@
+        return self._view.count
+
+    """
+    Operations
+    """
+
+    def reset(self, env_ids: Sequence[int] | None = None):
+        # reset the timestamps
+        super().reset(env_ids)
+        # resolve None
+        if env_ids is None:
+            env_ids = slice(None)
+        # reset accumulative data buffers
+        self._data.quat_w[env_ids] = 0.0
+        self._data.ang_vel_b[env_ids] = 0.0
+        self._data.lin_acc_b[env_ids] = 0.0
+
+    def update(self, dt: float, force_recompute: bool = False):
+        # save timestamp
+        self._dt = dt
+        # execute updating
+        super().update(dt, force_recompute)
+
+    """
+    Implementation.
+    """
+
+    def _initialize_impl(self):
+        """Initializes the sensor handles and internal buffers.
+
+        This function creates handles and registers the provided data types with the replicator registry to
+        be able to access the data from the sensor. It also initializes the internal buffers to store the data.
+
+        Raises:
+            RuntimeError: If the number of imu prims in the view does not match the number of environments.
+            RuntimeError: If the imu prim is not a RigidBodyPrim
+        """
+        # Initialize parent class
+        super()._initialize_impl()
+        # create simulation view
+        self._physics_sim_view = physx.create_simulation_view(self._backend)
+        self._physics_sim_view.set_subspace_roots("/")
+        # check if the prim at path is a rigid prim
+        prim = sim_utils.find_first_matching_prim(self.cfg.prim_path)
+        if prim is None:
+            raise RuntimeError(f"Failed to find a prim at path expression: {self.cfg.prim_path}")
+        # check if it is a RigidBody Prim
+        if prim.HasAPI(UsdPhysics.RigidBodyAPI):
+            self._view = self._physics_sim_view.create_rigid_body_view(self.cfg.prim_path.replace(".*", "*"))
+        else:
+            raise RuntimeError(f"Failed to find a RigidBodyAPI for the prim paths: {self.cfg.prim_path}")
+
+        # Create internal buffers
+        self._initialize_buffers_impl()
+
+    def _update_buffers_impl(self, env_ids: Sequence[int]):
+        """Fills the buffers of the sensor data."""
+        # check if self._dt is set (this is set in the update function)
+        if not hasattr(self, "_dt"):
+            raise RuntimeError(
+                "The update function must be called before the data buffers are accessed the first time."
+            )
+        # obtain the poses of the sensors
+        pos_w, quat_w = self._view.get_transforms()[env_ids].split([3, 4], dim=-1)
+        quat_w = math_utils.convert_quat(quat_w, to="wxyz")
+        # store the poses
+        # note: we clone here because the obtained tensors are read-only
+        self._data.pos_w[env_ids] = pos_w.clone() + math_utils.quat_rotate(quat_w.clone(), self._offset_pos)
+        self._data.quat_w[env_ids] = math_utils.quat_mul(quat_w.clone(), self._offset_quat)
```

Comment:
Is the clone operation here, needed? Don't think we do it for articulation or frame transformer?